schema_salad/ref_resolver.py

```diff
@@ -915,6 +918,14 @@ def validate_link(self, field, link, docid):
                 % (field, type(link).__name__))
         return link
 
+    def validate_ids(self, id):
```

Comment:
Unfortunately, if you call this method twice with the same parameters, it will raise an exception second time.  It should be possible to call a validation method twice and get the same result both times.

Instead of storing `all_doc_ids` as an attribute of `self`, it should be passed down as a parameter.